### PR TITLE
feat(tangle-dapp): Update Checkbox component

### LINKED ISSUE #2457: [TASK] Update Checkboxes component on webb-ui-components lib
### Description 
Update the UI of Checkbox component based on the new design in Figma

CleanShot 2024-07-25 at 21 06 34@2x

### PR DESCRIPTION
**Pull request by vutuanlinh2k2**
## Summary of changes

_Provide a detailed description of proposed changes._

- Update Checkbox component in `webb-ui-components` lib

### Proposed area of change

_Put an `x` in the boxes that apply._

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [x] `apps/tangle-dapp`
- [ ] `apps/testnet-leaderboard`
- [ ] `apps/faucet`
- [ ] `apps/zk-explorer`
- [x] `libs/webb-ui-components`

### Reference issue to close (if applicable)

_Specify any issues that can be closed from these changes (e.g. `Closes #233`)._

- Closes #2457 

### Screen Recording

_If possible provide a screen recording of proposed change._

https://github.com/user-attachments/assets/3e0203bb-87f4-470d-928f-ca0787c3c6dd

---

### Code Checklist

_Please be sure to add .stories documentation if any additions are made to `libs/webb-ui-components`._

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by monaiuu — APPROVED** [review]

**Comment by monaiuu** [comment]
On a separate note, I noticed in the screen recording there's inconsistencies here, might be good to quickly fix:
Screenshot 2024-07-30 at 3 38 28 PM

File: libs/webb-ui-components/src/components/CheckBox/Checkbox.tsx [hunk 1]
```diff
@@ -45,15 +45,16 @@ export const CheckBox: React.FC<CheckBoxProps> = (props) => {
...
   const inputClsx =
-    'form-checkbox peer transition-none bg-mono-0 w-[18px] h-[18px] rounded border border-mono-100 outline-none dark:bg-mono-180';
+    'form-checkbox peer transition-none bg-mono-0 w-[18px] h-[18px] rounded border-2 border-mono-100 outline-none dark:bg-mono-180 focus:ring-offset-0 focus:ring-0';
 
   const inputHoverClsx =
     'enabled:hover:shadow-sm enabled:hover:shadow-blue-10 dark:hover:shadow-none';
 
-  const inputCheckedClsx = 'checked:bg-blue-70 dark:checked:bg-blue-50';
+  const inputCheckedClsx =
+    'checked:!bg-blue-50 checked:hover:bg-blue-50 checked:active:!bg-blue-60 dark:checked:active:!bg-blue-40';
```
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/CheckBox/Checkbox.tsx:54]: I'm curious why you added `!important`, as it complicates component customization.

File: libs/webb-ui-components/src/components/CheckBox/Checkbox.tsx [hunk 2]
```diff
@@ -77,18 +78,34 @@ export const CheckBox: React.FC<CheckBoxProps> = (props) => {
...
-        type="checkbox"
-        className={mergedInputClsx}
-        checked={isChecked}
-        onChange={onChange}
-        disabled={isDisabled}
-        {...inputProps}
-      />
+    <label className={twMerge('inline-flex', wrapperClassName)}>
+      <div className="min-w-[28px]">
+        <div className="relative group min-h-[28px]">
```
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/CheckBox/Checkbox.tsx:83]: Can you use only 1 `div` here? Just

```
<div 
	className="group relative min-w-[28px] min-h-[28px]"
></div>
```

File: apps/tangle-dapp/components/ValidatorSelectionTable/ValidatorSelectionTable.tsx [hunk 3]
```diff
@@ -278,7 +278,7 @@ const ValidatorSelectionTable: FC<ValidatorSelectionTableProps> = ({
           ) : (
             <Table
               tableClassName={cx('[&_tr]:[overflow-anchor:_none]')}
-              thClassName="border-t-0 py-3 sticky top-0"
+              thClassName="z-10 border-t-0 py-3 sticky top-0"
```
Comment (AtelyPham) [inline apps/tangle-dapp/components/ValidatorSelectionTable/ValidatorSelectionTable.tsx:281]: Should the client need to set `z-index` with the `Checkbox` component, we should update the component to eliminate this requirement.

**Review by AtelyPham — COMMENTED** [review]
Left some comments.

**Review by yurixander — APPROVED** [review]

**Comment by vutuanlinh2k2** [comment]
> On a separate note, I noticed in the screen recording there's inconsistencies here, might be good to quickly fix: Screenshot 2024-07-30 at 3 38 28 PM

Fixed:
CleanShot 2024-07-30 at 21 00 41@2x

**Pull request merged**